src/SurfaceSchemes/surface_conditions.jl

```diff
@@ -0,0 +1,351 @@
+                if isnothing(q_flux)
+                    q_flux = FT(0)
+                else
+                    atmos.moisture_model isa DryModel && error(
+                        "q_flux cannot be specified when using a DryModel",
+                    )
+                end
+                ρ = TD.air_density(thermo_params, ts)
+                shf = θ_flux * ρ * TD.cp_m(thermo_params, ts)
+                lhf = q_flux * ρ * TD.latent_heat_vapor(thermo_params, ts)
+            end
+            if isnothing(gustiness)
+                buoyancy_flux = SF.compute_buoyancy_flux(
+                    surface_params,
+                    shf,
+                    lhf,
+                    interior_ts,
+                    ts,
+                    SF.FVScheme(),
+                )
+                gustiness = cbrt(buoyancy_flux * FT(1000))
+                # TODO: We are assuming that the average mixed layer depth is
+                # always 1000 meters. This needs to be adjusted for deep
+                # convective cases like TRMM.
+            end
+            isnothing(beta) || error(
+                "beta cannot be specified when surface fluxes are prescribed",
+            )
+            if isnothing(parameterization.ustar)
+                inputs = SF.Fluxes(
+                    interior_values,
+                    surface_values,
+                    shf,
+                    lhf,
+                    parameterization.z0m,
+                    parameterization.z0b,
+                    gustiness,
+                )
+            else
+                inputs = SF.FluxesAndFrictionVelocity(
+                    interior_values,
+                    surface_values,
+                    shf,
+                    lhf,
+                    parameterization.ustar,
+                    parameterization.z0m,
+                    parameterization.z0b,
+                    gustiness,
+                )
+            end
+        end
+    end
+
+    # The surface_conditions struct returned by SF does not store the values we
+    # need in ClimaAtmos (it does not contain the thermo state or any local
+    # geometry information), so we need to construct our own version of it.
+    (; ustar, L_MO, buoy_flux, ρτxz, ρτyz, shf, lhf, evaporation) =
+        SF.surface_conditions(surface_params, inputs)
+
+    C1 = Geometry.Covariant1Vector
+    C2 = Geometry.Covariant2Vector
+    C12 = Geometry.Covariant12Vector
+    CT3 = Geometry.Contravariant3Vector
+    surface_normal = CT3(unit_basis_vector_data(CT3, surface_local_geometry))
+    energy_flux = if atmos.energy_form isa PotentialTemperature
+        (; ρ_flux_θ = shf / TD.cp_m(thermo_params, ts) * surface_normal)
+    elseif atmos.energy_form isa TotalEnergy
+        if atmos.turbconv_model isa TC.EDMFModel
+            (;
+                ρ_flux_h_tot = (shf + lhf) * surface_normal,
+                ρ_flux_θ = shf / TD.cp_m(thermo_params, ts) * surface_normal,
```

Comment:
Where is `ρ_flux_θ` needed? I can't seem to find it?

Reply:
I think we can remove this now, right?

Reply:
Ok, we can't remove this because the old EDMF needs to call `θ_surface_bc`, which it needs `ρ_flux_θ` for. We'll remove this workaround when we delete the old EDMF.